Src/Sarif.PatternMatcher.Sdk/Crc32.cs

```diff
@@ -29,30 +37,46 @@ public static uint Calculate(byte[] buffer)
         }
 
         public static uint Calculate(uint checksum, byte[] buffer, int offset, int length)
+        {
+            return Calculate(checksum, buffer, offset, length, Crc32IEEETable);
+        }
+
+        public static uint Calculate(string text, uint[] crc32Table)
```

Comment:
>uint[] crc32Table [](http://example.com/codeflow?start=50&length=17)

do we allow user to pass in custom crc32table data or any uint array?

if not, I would create an enum for all kinds of crc32 tables, use it as the optional parameter, and change the public crc32 table properties to private since its encapsulated internal detail.

Reply:
changed